Git-commit: fdd7b863038aad64c89182134a40ebb669868d87
From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Tue, 2 Mar 2021 19:44:22 +0800
Subject: [PATCH] arm64: fix the compile error when CONFIG_NUMA is disabled
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Patch-mainline: Queued in openEuler repo, version 5.10.0-4.5.0
References: openEuler-21.03
Git-repo: https://gitee.com/openeuler/kernel.git

hulk inclusion
category: bugfix
bugzilla: 47994
CVE: NA

-------------------------------------------------------------------------

Fix the following compile error when CONFIG_NUMA is disabled.

arch/arm64/kernel/acpi.c: In function ‘acpi_map_cpu’:
arch/arm64/kernel/acpi.c:415:3: error: implicit declaration of function ‘numa_add_cpu’; did you mean ‘be64_add_cpu’? [-Werror=implicit-function-declaration]
   numa_add_cpu(cpu);
   ^~~~~~~~~~~~
   be64_add_cpu
arch/arm64/kernel/acpi.c: In function ‘acpi_unmap_cpu’:
arch/arm64/kernel/acpi.c:428:2: error: implicit declaration of function ‘numa_clear_node’; did you mean ‘pmd_clear_huge’? [-Werror=implicit-function-declaration]
  numa_clear_node(cpu);
  ^~~~~~~~~~~~~~~
  pmd_clear_huge

Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 arch/arm64/kernel/acpi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/kernel/acpi.c b/arch/arm64/kernel/acpi.c
index 0e59370bb215..b51ffac3b38d 100644
--- a/arch/arm64/kernel/acpi.c
+++ b/arch/arm64/kernel/acpi.c
@@ -31,6 +31,7 @@
 #include <asm/cputype.h>
 #include <asm/cpu_ops.h>
 #include <asm/daifflags.h>
+#include <asm/numa.h>
 #include <asm/smp_plat.h>
 
 int acpi_noirq = 1;		/* skip ACPI IRQ initialization */
-- 
2.31.1

